### PR TITLE
E2E: add temporary spec to test the WordPress.com Pro upgrade button.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR is the first in the series to lay the groundwork for implementing e2e coverage for the overhauled Plans feature.
This PR also includes a temporary stub spec meant to provide a baseline level of coverage for the Pro plan upgrade button while the fuller spec is being worked on.

Key changes:
1. removal of E2E bypass for the overhauled plans.
2. addition of new methods in the Plans POM to support interaction with the new plans grid.
3. addition of new e2e spec mirroring the addition of Plans Upgrade into the checkout cart.

Details:
1. currently all e2e runs from TeamCity are being redirected to show the legacy plans grid due to the bypass. In order to test the new plans grid, the bypass must be removed.
2. it has been decided to keep the Plans POM in one piece for the time being, therefore requiring new parameters, methods to support interactions with the new Plans grid and restructuring of some types.
3. the new e2e spec is essentially a mirror copy of the existing Plans: Upgrade spec which doesn't actually perform an upgrade; it clicks on a button in the Plans page to upgrade, validates the cart, then exits the checkout page. This spec will serve as a stand-in for now to provide minimal coverage for the time being while I work to spec out the critical code paths.

#### Testing instructions

Ensure the following:
  - [x] Gutenberg E2E (desktop)
  - [x] Gutenberg E2E (mobile)
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)
  - [x] Pre-Release E2E
  - [x] i18n E2E

Related to #62492

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/plans-page.ts [hunk 1]
```diff
@@ -97,31 +152,31 @@ export class PlansPage {
...
 
 	/**
 	 * Click a plan action button (on the plan cards on the "Plans" tab) based on expected plan name and button text.
 	 *
 	 * @param {object} param0 Object containing plan name and button text
-	 * @param {Plan} param0.plan Name of the plan (e.g. "Premium")
+	 * @param {LegacyPlans} param0.plan Name of the plan (e.g. "Premium")
 	 * @param {PlanActionButton} param0.buttonText Expected action button text (e.g. "Upgrade")
 	 */
 	async clickPlanActionButton( {
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/plans-page.ts:180]: I chose not to update the method name here:
- it adds a lot of unnecessary changes to the PR;
- if a developer tries to use this for the Pro plan, the lack of such option quickly becomes apparent.

File: packages/calypso-e2e/src/lib/pages/plans-page.ts [hunk 2]
```diff
@@ -17,31 +22,41 @@ const selectors = {
...
 	private page: Page;
+	private version: PlansGridVersion;
 
 	/**
 	 * Constructs an instance of the Plans POM.
 	 *
 	 * @param {Page} page Instance of the Playwright page
 	 */
-	constructor( page: Page ) {
+	constructor( page: Page, version: PlansGridVersion ) {
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/plans-page.ts:55]: I looked at keying off something else here, but the two-week ago me did not find anything useful to key off of.

The problem is partially due to the slower loading times for the Legacy plans grid which, frustratingly, is **not caught** by the `waitForLoadState('networkidle')`. 

Given the above, I tried to implement a `try/catch` to first look for the new plan, and fall back to the Legacy plan, but that introduces new problems:
- adds execution time if the page being shown is the legacy plans grid.
- what if the overhauled plans page takes just a bit longer than expected to load?

With all things considered, I felt it was important to require, explicitly, that the test writer specify what version of the Plans page they are intending to work with.

Comment (dpasque) [reply packages/calypso-e2e/src/lib/pages/plans-page.ts:55]: >With all things considered, I felt it was important to require, explicitly, that the test writer specify what version of the Plans page they are intending to work with.

Given our most recent discussion on the editor, I think this is a great approach! It's very deterministic -- we should know which page we expect. And if we don't get the one we expect, we should probably fail! So I think this will serve us very well 😄

File: packages/calypso-e2e/src/lib/pages/plans-page.ts [hunk 3]
```diff
@@ -17,31 +22,41 @@ const selectors = {
...
 		return `.plan-features__${ viewportSuffix } >> .plan-features__actions-button.is-${ plan.toLowerCase() }-plan:has-text("${ buttonText }")`;
 	},
 
-	// Plans view
-	plansGrid: '.plans-features-main',
+	// Legacy plans view
+	legacyPlansGrid: '.plans-features-main',
+
+	// Overhauled plans view
+	overhauledPlansGrid: 'div.plans > table',
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plans-page.ts]: Did you need this selector? Or is it for future use? I didn't see it used anywhere, so wanted to double check

File: packages/calypso-e2e/src/lib/pages/plans-page.ts [hunk 4]
```diff
@@ -17,31 +22,41 @@ const selectors = {
...
+	upgradeToProButton: 'th button.is-primary',
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plans-page.ts:33]: Can we not use button text due to potential i18n usage? Otherwise, I might opt for keying off the button text here, but that's a 100% optional suggestion! 👍

File: packages/calypso-e2e/src/lib/pages/plans-page.ts [hunk 5]
```diff
@@ -51,22 +66,62 @@ export class PlansPage {
...
+			locator.click(),
+		] );
+	}
+
+	/**
+	 * Clicks on the Show/Hide Plan Comparison button at the bottom of the table.
+	 *
+	 * @param {PlansComparisonAction} action Action to perform on the Plans comparison button.
+	 */
+	async clickPlanComparisonActionButton( action: PlansComparisonAction ): Promise< void > {
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plans-page.ts]: 100% optional suggestion: for here, what do you think of using that pattern you came up with where you have two methods which each guarantee their ending state? Like the [two methods here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts#L161)?

I've just been a huge fan of that design ever since you proposed it! 😄

**Review by dpasque — APPROVED** [review]
You did such a nice job weaving in the new coverage in an effective but non-disruptive way! 🎉 

I added a few optional comments - use them if you find the helpful! Otherwise, LGTM 🚀

File: packages/calypso-e2e/src/lib/pages/plans-page.ts (hunk 3 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/plans-page.ts]: Yep, this is an orphaned selector!

File: packages/calypso-e2e/src/lib/pages/plans-page.ts (hunk 4 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/plans-page.ts:33]: Right, I chose not to key off the button text for i18n concerns.

For English, there are at least two variants:
- `Start with Pro` (en-US)
- `Try Pro risk-free` (en-GB)

The user account (`simpleSiteFreePlanUser`) uses the en-GB language and I figured it would be too jarring for the test writer (whom is likely using en-US) to see an unexpectedly different string, so I chose to go with the CSS selector.
